directives/annotated.py

```diff
@@ -87,11 +92,27 @@ def depart_annotated_node(self, node):
     parsed_html = self.body  # extract generated feedback line
     self.body = env.redirect # restore original output
 
-    self.body.append(postprocess(u''.join(parsed_html), node['name']))
+    postprocessed_html = postprocess_annotation_tags(''.join(parsed_html), node['name'])
+    postprocessed_html = postprocess_inline_annotations(postprocessed_html, node['name'])
+    self.body.append(postprocessed_html)
 
     self.body.append("</div>\n")
 
-def postprocess(html, annotation_id):
+def postprocess_inline_annotations(html, annotated_section_id):
+    inline_anno_count = 0
+
+    def make_annotation_span(match):
+        nonlocal inline_anno_count
+        inline_anno_count += 1
+        annotation_text = match.group(1)
+        replacement = match.group(2)
+        replacement_attrib = ' data-replacement="' + replacement + '"' if replacement else ""
+        html_bits = (annotated_section_id, inline_anno_count, replacement_attrib, annotation_text)
+        return '<span class="codecomment comment-%s-%s"%s>%s</span>' % html_bits
```

Comment:
I have a feeling that the `annotation_text` and `replacement` values should be HTML escaped before writing them to HTML. If the RST content contains text like `5 > 3`, the `>` character could cause issues in the HTML if it is not escaped.

Reply:
https://docs.python.org/3/library/html.html#html.escape

Reply:
Unless I'm mistaken, that isn't an issue, because this code runs only after the contents of the `annotated::` directive have been processed into HTML (and escaped). At that point, `annotation_text` and `replacement` are part of that HTML output and this bit just relocates them within that putput. 

"Works fine when I tried it." ;-)

Reply:
I agree. Sphinx escapes the HTML here since the function processes the output of other nodes that are nested inside the annotated node. So no escaping should be added here.